README.md

```diff
@@ -1,68 +1,106 @@
 javaOptions += "-Djava.library.path=/usr/local/lib"
 ```
 
-You can use `pdal-native` dep in case you don't have installed JNI bindings and to avoid steps described above.
-Dependency contains bindings for `x86_64-darwin` and `x86_64-linux`, other versions are not supported yet.
-
-## PDAL-Scala (Scala 2.x)
 
+### Pdal-Scala (Scala 2.X)
 Scala API allows to build pipeline expressions instead of writing a raw JSON.
-
 ```scala
 // `<latest version>` refers to the version indicated by the badge above
 libraryDependencies ++= Seq(
-  "io.pdal" %% "pdal-scala" % "<latest version>", // scala core library
-  "io.pdal" %  "pdal-native" % "<latest version>" // jni bindings
+"io.pdal" %% "pdal-scala" % "<latest version>", // scala core library
+"io.pdal" %  "pdal-native" % "<latest version>" // jni bindings
 )
 ```
-
-Scala API covers PDAL 2.0.x, to use any custom DSL that is not covered by the 
+Scala API covers PDAL 2.0.x, to use any custom DSL that is not covered by the
 current Scala API you can use `RawExpr` type to build `Pipeline Expression`.
 
-## Code examples
+## Build
+Development purposes (including binaries) compilation:
+1. Install PDAL (using brew / package managers (unix) / build from sources / etc)
+2. Install sbt (using brew / package managers (unix))
+3. Build native libs `sbt native/nativeCompile` (optionally, binaries would be built during tests run)
+4. Run `sbt core/test` to run PDAL tests
+Only Java development purposes compilation:
+1. Provide `$LD_LIBRARY_PATH` or `$DYLD_LIBRARY_PATH`
+2. If you don't want to provide global variable you can pass `-Djava.library.path=<path>` into sbt:
+`./sbt -Djava.library.path=<path>`
+3. Set `PDAL_DEPEND_ON_NATIVE=false` (to disable `native` project build)
+4. Run `PDAL_DEPEND_ON_NATIVE=false sbt`
+Finally the possible command to launch and build PDAL JNI bindings could be:
+```bash
+# Including binaries build
+sbt
+```
+```bash
+# Java side development without binaries build
+PDAL_DEPEND_ON_NATIVE=false sbt -Djava.library.path=<path>
+```
 
-#### Demo project with examples
 
-JNI bindings basic usage examples can be found [here](./examples).
 
-### PDAL Core (Scala 2.x / 3.x)
 
+### Mac-OS ARM tutorial
+If you need to build your own `pdal-native` for `arm64-darwin`, the following is a temporary workaround:
+- If you are using `pdal` version `<= v2.4.x`, use `./sbt`. Otherwise, you need to install `sbt`: `brew install sbt`
+- Pay attention to your `pdal` installation: at the time of writing this, `conda` is the only working method, unless you build your own `pdal`. To install `pdal` with `conda`:
+```
+brew install miniconda
+conda create --name pdal_env python=3.9 # to create an environment separate from your system's
+conda activate pdal_env
+conda install -c conda-forge pdal
+pdal --version # to check the installation
+conda env export --from-history > pdal_env.yml # if you need to export the env for collaborative use
+conda deactivate # to exit the conda env
```

Comment:
That's not really a part of the repo. There's more than a single way to install PDAL "properly", env is up to user. We could mention smth like a "known issue on MAC" / or "PDAL installed via conda is tested"

The only requirement is that PDAL should be fully functional.

i.e. 

* brew works already: https://github.com/Homebrew/homebrew-core/blob/master/Formula/p/pdal.rb#L4; tested it today; 
* another one that works is installing and building PDAL from sources (always an hardcore option)

Reply:
the problem of brew is that it works 1 time out of 10 and building PDAL from sources is not really easy for users that are not expert. Also, with brew is not that easy to install things in a separate env.
 I moved the section under well known issues, for me it's always been a issue: for example i had pdal correctly installed through nix but the compilation of the natives was not working

Reply:
Could write that there may be issues with other enviornments, hard really to say; we should not try to describe all possible issues and PDAL installation methods, we can even add a link to PDAL docs https://pdal.io/en/2.6.0/quickstart.html and rm the rest.